crates/cli/src/config.rs

```diff
@@ -14,6 +25,21 @@ pub struct ServerConfig {
 }
 
 impl ServerConfig {
+    /// Generate a new [Table] representing this [ServerConfig].
+    pub fn as_table(&self) -> Table {
+        let mut table = Table::new();
+        Self::update_table(&mut table, self);
+        table
+    }
+
+    /// Update an existing [Table] with the values of a [ServerConfig].
```

Comment:
```suggestion
    /// Update an existing [`Table`] with the values of a [`ServerConfig`].
```

This needs a paragraph describing its specific behavior. What data from the existing `Table` are preserved? What are overwritten?